[BUG]: BicepDeploy@0 requires a parameter file

### New issue checklist

- [x] I searched for [existing GitHub issues](https://github.com/microsoft/azure-pipelines-tasks/issues)
- [x] I read [pipeline troubleshooting guide](https://docs.microsoft.com/vsts/build-release/actions/troubleshooting)
- [x] I checked [how to collect logs](https://learn.microsoft.com/azure/devops/pipelines/troubleshooting/review-logs?view=azure-devops)

### Task name

BicepDeploy

### Task version

0.271.0

### Issue Description

I tried to use this task to deploy a Bicep file.  This was purely for testing purposes and the Bicep used all hard-coded values and did not specify any parameters.  Therefore, I omitted the `parametersFile` key from the task's inputs.  However, the task failed with this error:

```
##[error]OperationFailed
##[error]Unsupported parameters file type: /home/vsts/work/1/s
```

### Environment type (Please select at least one enviroment where you face this issue)

- [ ] Self-Hosted
- [x] Microsoft Hosted
- [ ] VMSS Pool
- [ ] Container

### Azure DevOps Server type

dev.azure.com (formerly visualstudio.com)

### Azure DevOps Server Version (if applicable)

_No response_

### Operation system

Ubuntu-latest

### Relevant log output

```shell
n/a
```

### Full task logs with system.debug enabled

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>


### Repro steps

```yml

```

--- a/Tasks/BicepDeployV0/README.md
+++ b/Tasks/BicepDeployV0/README.md
@@ -178,12 +178,13 @@ For detailed guidance, refer to the [Bicep Outputs documentation](https://learn.
 
 **How Outputs Become Available**
 
-After the deployment completes successfully, all outputs are automatically converted to Azure Pipelines variables with the same names as defined in your Bicep template. These variables are created at the pipeline level and can be directly referenced in subsequent tasks using the standard variable syntax: `$(outputName)`.
+After the deployment completes successfully, all outputs are set as output variables for the task step. To reference them in subsequent tasks, give the BicepDeploy step a `name` and use the syntax `$(stepName.outputName)`.
 
 ### Accessing Outputs in PowerShell
 
 ```yaml
 - task: BicepDeploy@0
+  name: deploy
   displayName: 'Deploy Infrastructure'
   inputs:
     azureResourceManagerConnection: 'Azure-Connection'
@@ -196,14 +197,15 @@ After the deployment completes successfully, all outputs are automatically conve
   inputs:
     targetType: 'inline'
     script: |
-      Write-Host "Storage Account Name: $(storageAccountName)"
-      Write-Host "Web App URL: $(webAppUrl)"
+      Write-Host "Storage Account Name: $(deploy.storageAccountName)"
+      Write-Host "Web App URL: $(deploy.webAppUrl)"
 ```
 
 ### Accessing Outputs in Bash
 
 ```yaml
 - task: BicepDeploy@0
+  name: deploy
   displayName: 'Deploy Infrastructure'
   inputs:
     azureResourceManagerConnection: 'Azure-Connection'
@@ -216,8 +218,8 @@ After the deployment completes successfully, all outputs are automatically conve
   inputs:
     targetType: 'inline'
     script: |
-      echo "Storage Account Name: $(storageAccountName)"
-      echo "Web App URL: $(webAppUrl)"
+      echo "Storage Account Name: $(deploy.storageAccountName)"
+      echo "Web App URL: $(deploy.webAppUrl)"
 ```
 
 ## Feedback and Support

--- a/Tasks/BicepDeployV0/Strings/resources.resjson/en-US/resources.resjson
+++ b/Tasks/BicepDeployV0/Strings/resources.resjson/en-US/resources.resjson
@@ -79,6 +79,7 @@
   "loc.messages.LocationRequired": "Location is required",
   "loc.messages.TemplateFileRequired": "Template file is required",
   "loc.messages.UnsupportedTemplateFile": "Unsupported template file type: %s",
+  "loc.messages.ParametersFileNotFound": "Parameters file not found: %s",
   "loc.messages.UnsupportedParametersFile": "Unsupported parameters file type: %s",
   "loc.messages.FailedToDetermineScope": "Failed to determine deployment scope from Bicep file.",
   "loc.messages.FailedToResolveBicepVersion": "Failed to resolve Bicep version from download URL: %s",

--- a/Tasks/BicepDeployV0/Strings/resources.resjson/en-US/resources.resjson
+++ b/Tasks/BicepDeployV0/Strings/resources.resjson/en-US/resources.resjson
@@ -85,6 +85,7 @@
   "loc.messages.FailedToResolveBicepVersion": "Failed to resolve Bicep version from download URL: %s",
   "loc.messages.CreateFailed": "Create failed",
   "loc.messages.ValidationFailed": "Validation failed",
+  "loc.messages.WhatIfFailed": "What-if failed",
   "loc.messages.OperationFailed": "Operation failed",
   "loc.messages.RequestFailedCorrelation": "Request failed. CorrelationId: %s",
   "loc.messages.InvalidChangeType": "Invalid change type: %s",

--- a/Tasks/BicepDeployV0/Tests/L0.ts
+++ b/Tasks/BicepDeployV0/Tests/L0.ts
@@ -133,6 +133,16 @@ describe("deployments tests", function() {
       { name: 'objectOutput', value: '{"key1":"value1","key2":"value2"}' },
     ], this.test!.title);
   });
+
+  it("runs create with no parameters file", async function() {
+    let tp: string = path.join(__dirname, "deploymentsCreateNoParamFile.js");
+
+    let tr: ttm.MockTestRunner = new ttm.MockTestRunner(tp);
+
+    await tr.runAsync();
+
+    assertSucceeded(tr, this.test!.title);
+  });
 });
 
 describe("stacks tests", function() {

--- a/Tasks/BicepDeployV0/Tests/deploymentsCreateNoParamFile.ts
+++ b/Tasks/BicepDeployV0/Tests/deploymentsCreateNoParamFile.ts
@@ -1,0 +1,35 @@
+// Import and configure rewiremock BEFORE anything else loads
+const rewiremock = require('rewiremock/node');
+
+import * as path from 'path';
+import { createDeploymentMockRunner } from './utils';
+
+// Mock deploymentCreate to return successful deployment (no outputs needed for this test)
+const mockDeploymentCreate = async () => {
+  return {
+    properties: {
+      outputs: {}
+    }
+  };
+};
+
+// Enable rewiremock globally
+rewiremock.enable();
+
+// Mock the deployments module
+rewiremock('@azure/bicep-deploy-common/deployments')
+  .callThrough()
+  .with({ deploymentCreate: mockDeploymentCreate });
+
+let tr = createDeploymentMockRunner(rewiremock);
+
+tr.setInput('operation', 'create');
+tr.setInput('name', 'test-create-no-param');
+tr.setInput('templateFile', path.join(__dirname, 'files', 'basic', 'main.bicep'));
+// parametersFile is intentionally NOT set — simulates user leaving it empty.
+// In ADO, filePath-type inputs resolve empty values to Build.SourcesDirectory,
+// so isFilePathSupplied should return false and the task should succeed without a parameters file.
+
+tr.run();
+
+rewiremock.disable();

--- a/Tasks/BicepDeployV0/logging.ts
+++ b/Tasks/BicepDeployV0/logging.ts
@@ -55,6 +55,8 @@ export const errorMessageConfig: ErrorMessageConfig = {
     failedToDetermineScope: tl.loc('FailedToDetermineScope'),
     
     // File errors
+    parametersFileNotFound: (parametersFile: string) =>
+        tl.loc('ParametersFileNotFound', parametersFile),
     unsupportedParametersFile: (parametersFile: string) => 
         tl.loc('UnsupportedParametersFile', parametersFile),
     unsupportedTemplateFile: (templateFile: string) => 

--- a/Tasks/BicepDeployV0/logging.ts
+++ b/Tasks/BicepDeployV0/logging.ts
@@ -34,6 +34,7 @@ export const errorMessageConfig: ErrorMessageConfig = {
     // Handler errors
     createFailed: tl.loc('CreateFailed'),
     validationFailed: tl.loc('ValidationFailed'),
+    whatIfFailed: tl.loc('WhatIfFailed'),
     operationFailed: tl.loc('OperationFailed'),
     requestFailedCorrelation: (correlationId: string) => 
         tl.loc('RequestFailedCorrelation', correlationId),

--- a/Tasks/BicepDeployV0/package.json
+++ b/Tasks/BicepDeployV0/package.json
@@ -16,7 +16,7 @@
   },
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
-    "@azure/bicep-deploy-common": "^0.0.4-dev",
+    "@azure/bicep-deploy-common": "^0.0.5",
     "@types/mocha": "^5.2.7",
     "@types/node": "^24.10.0",
     "@types/q": "^1.5.8",

--- a/Tasks/BicepDeployV0/package.json
+++ b/Tasks/BicepDeployV0/package.json
@@ -16,7 +16,7 @@
   },
   "homepage": "https://github.com/Microsoft/azure-pipelines-tasks#readme",
   "dependencies": {
-    "@azure/bicep-deploy-common": "^0.0.5",
+    "@azure/bicep-deploy-common": "^0.0.6",
     "@types/mocha": "^5.2.7",
     "@types/node": "^24.10.0",
     "@types/q": "^1.5.8",

--- a/Tasks/BicepDeployV0/task.json
+++ b/Tasks/BicepDeployV0/task.json
@@ -10,7 +10,7 @@
   "author": "Microsoft Corporation",
   "version": {
     "Major": 0,
-    "Minor": 271,
+    "Minor": 273,
     "Patch": 0
   },
   "minimumAgentVersion": "2.144.0",
@@ -383,6 +383,7 @@
     "LocationRequired": "Location is required",
     "TemplateFileRequired": "Template file is required",
     "UnsupportedTemplateFile": "Unsupported template file type: %s",
+    "ParametersFileNotFound": "Parameters file not found: %s",
     "UnsupportedParametersFile": "Unsupported parameters file type: %s",
     "FailedToDetermineScope": "Failed to determine deployment scope from Bicep file.",
     "FailedToResolveBicepVersion": "Failed to resolve Bicep version from download URL: %s",

--- a/Tasks/BicepDeployV0/task.json
+++ b/Tasks/BicepDeployV0/task.json
@@ -389,6 +389,7 @@
     "FailedToResolveBicepVersion": "Failed to resolve Bicep version from download URL: %s",
     "CreateFailed": "Create failed",
     "ValidationFailed": "Validation failed",
+    "WhatIfFailed": "What-if failed",
     "OperationFailed": "Operation failed",
     "RequestFailedCorrelation": "Request failed. CorrelationId: %s",
     "InvalidChangeType": "Invalid change type: %s",

--- a/Tasks/BicepDeployV0/task.loc.json
+++ b/Tasks/BicepDeployV0/task.loc.json
@@ -392,6 +392,7 @@
     "FailedToResolveBicepVersion": "ms-resource:loc.messages.FailedToResolveBicepVersion",
     "CreateFailed": "ms-resource:loc.messages.CreateFailed",
     "ValidationFailed": "ms-resource:loc.messages.ValidationFailed",
+    "WhatIfFailed": "ms-resource:loc.messages.WhatIfFailed",
     "OperationFailed": "ms-resource:loc.messages.OperationFailed",
     "RequestFailedCorrelation": "ms-resource:loc.messages.RequestFailedCorrelation",
     "InvalidChangeType": "ms-resource:loc.messages.InvalidChangeType",

--- a/Tasks/BicepDeployV0/task.loc.json
+++ b/Tasks/BicepDeployV0/task.loc.json
@@ -13,7 +13,7 @@
   "author": "Microsoft Corporation",
   "version": {
     "Major": 0,
-    "Minor": 271,
+    "Minor": 273,
     "Patch": 0
   },
   "minimumAgentVersion": "2.144.0",
@@ -386,6 +386,7 @@
     "LocationRequired": "ms-resource:loc.messages.LocationRequired",
     "TemplateFileRequired": "ms-resource:loc.messages.TemplateFileRequired",
     "UnsupportedTemplateFile": "ms-resource:loc.messages.UnsupportedTemplateFile",
+    "ParametersFileNotFound": "ms-resource:loc.messages.ParametersFileNotFound",
     "UnsupportedParametersFile": "ms-resource:loc.messages.UnsupportedParametersFile",
     "FailedToDetermineScope": "ms-resource:loc.messages.FailedToDetermineScope",
     "FailedToResolveBicepVersion": "ms-resource:loc.messages.FailedToResolveBicepVersion",

--- a/Tasks/BicepDeployV0/taskIO.ts
+++ b/Tasks/BicepDeployV0/taskIO.ts
@@ -5,6 +5,10 @@ export class TaskInputReader implements InputReader {
     getInput(inputName: string): string | undefined {
         return tl.getInput(inputName, false);
     }
+
+    isFilePathSupplied(inputName: string): boolean {
+        return tl.filePathSupplied(inputName);
+    }
 }
 
 export class TaskOutputSetter implements OutputSetter {

